### PR TITLE
Composer: new docs page explaining the different authentication tokens

### PR DESCRIPTION
**Pull request by glaubinix**

### CODE REVIEW

**Review by zanbaldwin — APPROVED** [review]

**Review by naderman — CHANGES_REQUESTED** [review]

File: docs/composer-authentication.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Tokens created in the organization settings do not grant you access to any of your subrepositories and tokens created in
+a subrepository only grant you access to the subrepository the token was created in.
+
+#### Read-only vs update tokens
+There are two kinds of organization authentication tokens, read-only and update tokens.
+Read-only tokens aren't suitable to run `composer update` as they do not allow automatic mirroring for new packages.
+If you need to run `composer update` then either use a token with update access or your personal access token.
+Please note that you will be charged for authentication tokens with update access as if they were user accounts.
+
+### Private Packagist for Vendors customer tokens
```
[the remark this answers is not in the document]
Comment (glaubinix) [reply docs/composer-authentication.md:27]: ah yes, I see how this can be confusing

**Comment by glaubinix** [comment]
Thank you!

File: docs/composer-authentication.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* `packagist_ort_`: Organization tokens with read-only access
+* `packagist_out_`: Organization tokens with update access
+* `packagist_uut_`: User tokens with update access
+
+This format doesn't apply to authentication tokens generated for Private Packagist for Vendors customers and older tokens that haven't been regenerated recently.
+These tokens only consist of the 60 hexadecimal character random part.
+
+### How to calculate the checksum
+To calculate the checksum of a token, calculate the CRC32 checksum using the prefix and random part, convert the number to hexadecimal
+and asserting that the checksum is exactly eight characters long.
```
[the remark this answers is not in the document]
Comment (glaubinix) [reply docs/composer-authentication.md]: @naderman the part `using the prefix and random part convert the number to hexadecimal` reads a bit weird. Is this really correct?

[the remark this answers is not in the document]
Comment (glaubinix) [reply docs/composer-authentication.md]: Did you mean `random part. Convert the number` to split this into two sentences?

[the remark this answers is not in the document]
Comment (naderman) [reply docs/composer-authentication.md]:
```suggestion
To calculate the checksum of a token, calculate the CRC32 checksum using the prefix and random part, and convert the number to hexadecimal while padding it to exactly 8 characters.
```

[the remark this answers is not in the document]
Comment (glaubinix) [reply docs/composer-authentication.md]: ok, fixed

**Review by naderman — APPROVED** [review]

**Pull request merged**